### ISSUE TITLE
The "Dismiss" notification button fix

### ISSUE THREAD

**Issue by dagayaru** [nextcloud/client#117]
It's about https://github.com/nextcloud/client/commit/75676f883092426c7f669b65f96f01e36f0a1bc2 and the corresponding owncloud/client commit.

The `json.value("notification_id")` is `double` and `json.value("notification_id").toString()` is `""` and the server returns 404.

So, it should be

```
diff --git a/client_theming/client/src/gui/servernotificationhandler.cpp b/client_theming/client/src/gui/servernotificationhandler.cpp
index 72b65b2..fdf7b4d 100644
--- a/client_theming/client/src/gui/servernotificationhandler.cpp
+++ b/client_theming/client/src/gui/servernotificationhandler.cpp
@@ -101,7 +101,7 @@ void ServerNotificationHandler::slotNotificationsReceived(const QJsonDocument &j
         // https://github.com/owncloud/notifications/blob/master/docs/ocs-endpoint-v1.md#deleting-a-notification-for-a-user
         ActivityLink al;
         al._label = tr("Dismiss");
-        al._link  = Utility::concatUrlPath(ai->account()->url(), notificationsPath + "/" + json.value("notification_id").toString()).toString();
+        al._link  = Utility::concatUrlPath(ai->account()->url(), notificationsPath + "/" + QString::number(a._id)).toString();
         al._verb  = "DELETE";
         al._isPrimary = false;
         a._links.append(al);
```

**Comment by nickvergessen** [comment]
I always wondered why the dismiss did not work 👍

**Comment by rullzer** [comment]
@camilasan could you look into this?

**Comment by camilasan** [comment]
Thanks @dagayaru. 
I created a PR: #133

**Comment by rullzer** [comment]
Merged with #133
